### ISSUE TITLE
Drop CoffeeScript

### ISSUE THREAD

**Issue by lipis** [wireapp/wire-webapp#891]
We decided to slowly start moving away from CoffeeScript to JavaScript! As this step is impossible to do it in one go, here is the list of things needs to be done before we sign it off.

### Tools that could help

- [js2.coffee](http://js2.coffee/)
- [lebab.io](http://lebab.io/try-it)
- [decaffeinate](https://github.com/decaffeinate/decaffeinate)

### Things to do

- [x] Support JavaScript, along with CoffeeScript in the build scripts (#883)
- [x] Add ESLint to build pipeline (#889) 
- [x] `app/`
  - [x] `script/`
    - [x] `config.coffee` (#888)
    - [x] `announce/` (#899)
    - [x] `assets/` (#912)
    - [x] `audio/` (#925)
    - [x] `auth/` (#947)
    - [x] `bot/`(#941)
    - [x] `cache/` (#946)
    - [x] `calling/` (#1125)
      - [x] Remove deprecated calling v2 code (#1340)
    - [x] `client/` (#976)
    - [x] `components/` (#1207)
    - [x] `connect/` (#1030)
    - [x] `conversation/` (#1172)
    - [x] `cryptography/` (#1048)
    - [x] `entity/` (#1195)
    - [x] `entity/message` (#1097)
    - [x] `ephemeral/` (#950)
    - [x] `event/` (#1078)
    - [x] `extension/` (#1081)
    - [x] `links/` (#963)
    - [x] `localization/` (#1036)
    - [x] `location/` (#951)
    - [x] `main/` (#1084)
    - [x] `media/` (#1065)
    - [x] `message/` (#1086)
    - [x] `properties/` (#1079)
    - [x] `search/` (#1082)
    - [x] `service/` (#1080)
    - [x] `storage/` (#1083)
    - [x] `system_notification/` (#1052)
    - [x] `telemetry/` (#1089)
    - [x] `tracking/` (#1085)
    - [x] `ui/` (#1120)
    - [x] `user/` (#1138)
    - [x] `util/` (#1114)
    - [x] `view_model/` (#1236)
      - [x] `view_model/bindings/` (#1222)
      - [x] `view_model/content/` (#1235)
      - [x] `view_model/list/` (#1233)
  - [x] `worker` (#1374)
- [x] `test/api`
- [x] `test/unit_tests/` (#1201)
  - [x] `announce/` (#1201)
  - [x] `assets/` (#1201)
  - [x] `audio/` (#1201)
  - [x] `auth/` (#1201)
  - [x] `cache/` (#1201)
  - [x] `calling/` (#1201)
  - [x] `client/` (#1201)
  - [x] `connect/` (#1201)
  - [x] `conversation/` (#1201)
  - [x] `cryptography/` (#1201)
  - [x] `entity/` (#1201)
  - [x] `event/` (#1201)
  - [x] `extension/` (#1201)
  - [x] `links/` (#1201)
  - [x] `localization/` (#1036)
  - [x] `location/` (#951)
  - [x] `media/` (#1201)
  - [x] `message/` (#1201)
  - [x] `search/` (#1201)
  - [x] `service/` (#1201)
  - [x] `storage/` (#1201)
  - [x] `system_notification/` (#1201)
  - [x] `tracking/` (#1201)
  - [x] `ui/` (#1201)
  - [x] `user/` (#1201)
  - [x] `util/`  (#1201)
  - [x] `view_model/` (#1201)
- [x] `Gruntfile.coffee` (#885)
- [x] `grunt/config/` (#1376)
- [x] `grunt/tasks/` (#1376)
- [x] Make sure there are no `*.coffee` files in the project
- [x] Remove build tasks `package.json`
- [x] Remove the CoffeeScript dependency from `package.json`

**Comment by bennyn** [comment]
Test files also need to get transpiled.

**Comment by balupton** [comment]
Not sure what the reasoning is, but CoffeeScript v2 will be out soon: https://github.com/coffeescript6/discuss/issues/71

Over at Bevry, we moved from CoffeeScript to ESNext for all our things throughout 2016. Our reasoning is at: https://discuss.bevry.me/t/move-from-coffeescript-to-es6/30/1 - perhaps will have some useful resources for your own move.

**Comment by herrmannplatz** [comment]
Thanks for sharing this. Haven't followed the whole v2 discussion. Was quite surprised to see that they actually started working on this finally. But this doesn't change our decision. What you mentioned is similar to what we discussed. Besides that some more reasons from our site are:

- We only support recent browser versions which can handle es6. So we can get rid of a compile step and start writing plain es6. Using babel on top is also easier with this setup. 
- We had some nasty bugs because of implicit returns
- Interoperability with other frameworks and tooling. People putting less and less effort to support CoffeeScript (e.g React esp. JSX)

To sum it up. CoffeeScript introduced some really helpful concepts and made coding more convenient, but JavaScript nowadays offers so much more.

**Comment by lipis** [comment]
<img width="996" alt="screen shot 2017-04-06 at 16 11 51" src="https://cloud.githubusercontent.com/assets/125676/24758816/c5e813ca-1ae3-11e7-8294-ee9824e2cac8.png">

**Comment by lipis** [comment]
<img width="990" alt="screen shot 2017-04-07 at 22 42 11" src="https://cloud.githubusercontent.com/assets/125676/24819093/80dcc342-1be3-11e7-8349-ac1f8a891b93.png">

**Comment by lipis** [comment]
<img width="989" alt="screen shot 2017-04-12 at 12 05 59" src="https://cloud.githubusercontent.com/assets/125676/24952622/6e940036-1f78-11e7-87a5-8fd72f8197b3.png">

**Comment by lipis** [comment]
<img width="992" alt="screen shot 2017-04-12 at 13 59 22" src="https://cloud.githubusercontent.com/assets/125676/24956530/4d31e4ac-1f88-11e7-8c41-2752ce45dc91.png">

**Comment by lipis** [comment]
<img width="986" alt="screen shot 2017-04-18 at 20 04 04" src="https://cloud.githubusercontent.com/assets/125676/25143194/3c4bfca6-2472-11e7-8e73-021aeb859dd4.png">

**Comment by gregor** [comment]
![coffeescript](https://cloud.githubusercontent.com/assets/10243309/25225393/89cc3a0e-25c1-11e7-988b-4f8c422c14dd.png)

**Comment by bennycode** [comment]
![unbenannt](https://cloud.githubusercontent.com/assets/469989/25229442/8a59fae2-25d0-11e7-8c32-8cd9fe38c7c1.PNG)

**Comment by bennycode** [comment]
![Screenshot](https://cloud.githubusercontent.com/assets/469989/25383351/09c89ac4-29bc-11e7-8c8b-c2adfec04344.PNG)

**Comment by gregor** [comment]
![stuff](https://cloud.githubusercontent.com/assets/10243309/25445422/af0e238a-2aae-11e7-92c7-a64095384135.png)

Finally 😄

**Comment by bennycode** [comment]
![unbenannt](https://cloud.githubusercontent.com/assets/469989/25655403/0740e0b0-2ff5-11e7-8027-1882d6f89b21.PNG)

**Comment by bennycode** [comment]
<img width="990" alt="screen shot 2017-05-09 at 22 30 05" src="https://cloud.githubusercontent.com/assets/125676/25871214/07228d1a-3507-11e7-806e-85c263cb1bba.png">

**Comment by lipis** [comment]
<img width="988" alt="screen shot 2017-05-09 at 22 24 42" src="https://cloud.githubusercontent.com/assets/125676/25870972/3e03dc7c-3506-11e7-9760-7c160b13148e.png">

**Comment by gregor** [comment]
![js](https://cloud.githubusercontent.com/assets/10243309/25898378/32f24c8e-358c-11e7-8ce2-da1bf57190d4.png)

**Comment by lipis** [comment]
<img width="985" alt="screen shot 2017-05-11 at 18 45 19" src="https://cloud.githubusercontent.com/assets/125676/25961236/e70b0a7e-3679-11e7-9396-62fed6677b47.png">

**Comment by gregor** [comment]
![js](https://cloud.githubusercontent.com/assets/10243309/26050103/bfbcf8c4-395d-11e7-92f1-ef7df18a301a.png)

**Comment by gregor** [comment]
![js](https://cloud.githubusercontent.com/assets/10243309/26071790/90e77320-39a9-11e7-9e10-c0d8f40745b8.png)

Another milestone passed 🎉

**Comment by gregor** [comment]
![js](https://cloud.githubusercontent.com/assets/10243309/26346444/8f4d1dc8-3fa6-11e7-84aa-3ba5fafae942.png)

**Comment by gregor** [comment]
![js](https://user-images.githubusercontent.com/10243309/26888753-4702cd3a-4bac-11e7-8c8d-9e34aeeb5098.png)

Bääm

**Comment by herrmannplatz** [comment]
<img width="997" alt="bildschirmfoto 2017-06-12 um 17 44 54" src="https://user-images.githubusercontent.com/174685/27042347-f3c7e63e-4f96-11e7-9f8b-40fc8150132b.png">

**Comment by herrmannplatz** [comment]
DONE!

**Comment by lipis** [comment]
KAPUT!

**Comment by herrmannplatz** [comment]
Now i should start drinking less coffee -_-'

**Comment by lipis** [comment]
but but but Java is Coffee 🗡

**Comment by gregor** [comment]
Let's have a celebration tonight 🎉
